spark-cassandra-connector/src/test/scala/com/datastax/spark/connector/writer/RateLimiterSpec.scala

```diff
@@ -4,44 +4,36 @@ import org.scalatest.{FlatSpec, Matchers}
 
 class RateLimiterSpec extends FlatSpec with Matchers {
 
-  // Warning, this test is timing-based.
-  // It may fail if run on a heavily loaded system with unpredictable performance.
-  // Increase allowedSystemInducedPause if you encounter any problems:
-  val allowedSystemInducedPause = 250 // ms
-
   "RateLimiter" should "not cause delays if rate is not exceeded" in {
-    val limiter = new RateLimiter(Long.MaxValue, 1000)
-    val start1 = System.currentTimeMillis()
-    for (i <- 1 to 1000000)
-      limiter.maybeSleep(0)
-    val end1 = System.currentTimeMillis()
-    val referenceElapsed = end1 - start1
-
-    val start2 = System.currentTimeMillis()
-    for (i <- 1 to 1000000)
-      limiter.maybeSleep(1000)
-    val end2 = System.currentTimeMillis()
-    val testElapsed = end2 - start2
-    testElapsed should be <= referenceElapsed + allowedSystemInducedPause
+    var now: Long = 0
+    def sleep(delay: Long) = throw new Exception("Sleep must never be called in this test")
```

Comment:
why not using ScalaMocks here instead? It is cleaner to me - you can check than assert that no interaction happened.